jukebox/src/main/webapp/js/youtubePlayerScript.js

```diff
@@ -7,19 +7,26 @@
 
     var player;
     var display_container = document.getElementById("display_container");
+    var isVideoPlaying = true; //Keeps track of whether video is playing; ignores user pauses
 
     //Finds the youtube video and creates an iframe
     function getYouTube(vidID, nextSong) {
         //Case where there are no videos
         if(vidID == NO_VIDEO){
             display_container.innerHTML = `No songs in queue.`;
+            //If player has been created, stop the video
+            if (player) {
+                stopVideo();
+            }
+            isVideoPlaying = false;
             return;
         }
-
+        isVideoPlaying = true;
+        display_container.textContent = `Currently Playing...`;
         //New song, so create a new iframe
         if(!nextSong){
             console.log("test"+ vidID);
-            display_container.innerHTML = `Currently Playing ...`;
+            
             player = new YT.Player('player', {
```

Comment:
can we always have this initialized so that we don't need to worry about checking whether its created or not all the time? that way, we can focus on whether "isVideoPlaying" is true or not

Reply:
We could probably split getYouTube into two separate methods where the first one initializes the iframe while the other one updates it to the next song.

Reply:
I've been looking for a possible way to have the player always initialized, but it seems to me that Jeremy's original code is the best solution. The main problem is that the youtube player can only be initialized with a video ID. This means that the player can't be loaded before the first song is added to the queue, because there's no video ID. I tried initializing the player upon loading the page, and omitting the video ID variable. It works as normal, but the user is greeted by an error whenever they create their room/have an empty queue.

(this [post](https://stackoverflow.com/questions/12508301/init-youtube-iframe-player-without-video-id)'s solution is the same one I tried, but the response claiming it's outdated seems to be true).

<img width="1435" alt="Screen Shot 2020-05-18 at 5 12 01 PM" src="https://user-images.githubusercontent.com/46883117/82270783-bbffc880-992a-11ea-9164-e21dcbb85e82.png">